lib/readline.js

```diff
@@ -45,3 +46,34 @@ module.exports = {
   emitKeypressEvents,
   moveCursor
 };
+
+let promises;
+
+Object.defineProperties(module.exports, {
+  promises: {
+    configurable: true,
+    enumerable: false,
+    get() {
+      if (promises === undefined) {
+        const {
+          Interface,
+          createInterface
+        } = require('internal/readline/promises');
+
+        promises = {
+          Interface,
+          clearLine,
+          clearScreenDown,
+          createInterface,
+          cursorTo,
+          emitKeypressEvents,
+          moveCursor
```

Comment:
I think all of these, except for `Interface` and `emitKeypressEvents`, are async operations because they write to the underlying stream. Should we account for that by making them return Promises?

Reply:
@addaleax I've been thinking about this. I'm a little torn. Returning a `Promise` does make sense. But why doesn't the current readline implementation consider the `write()` callback at all? The return value of `write()` is also not considered in either implementation currently.

Reply:
> But why doesn't the current readline implementation consider the `write()` callback at all? The return value of `write()` is also not considered in either implementation currently.

Because inconsistent API design is one of Node’s big weaknesses, sadly… (And probably also because Node follows a “small writes work synchronously 99 % of the time” model, with all the pain that comes with the other 1 %).

But yeah, I’d prefer not repeating past mistakes. The return value is something I hadn’t considered yet, to be honest, but I’d probably give the write callback preference?